frontend/components/Home/Dashboard/StudentCourses.tsx

```diff
@@ -5,36 +5,64 @@ import Alert from "@material-ui/lab/Alert";
 import CourseCard from "./Cards/CourseCard";
 import AddCard from "./Cards/AddCard";
 import ModalAddStudentCourse from "./Modals/ModalAddStudentCourse";
-import { Course, mutateFunction, UserMembership } from "../../../types";
+import ModalLeaveStudentCourse from "./Modals/ModalLeaveStudentCourse";
+import { mutateFunction, Toast, UserMembership } from "../../../types";
 
 interface StudentCoursesProps {
-    courses: Course[];
+    memberships: UserMembership[];
     mutate: mutateFunction<UserMembership[]>;
 }
-
 const StudentCourses = (props: StudentCoursesProps) => {
-    const { mutate, courses } = props;
+    const { mutate, memberships } = props;
     /* STATE */
     const [open, setOpen] = useState(false);
-    const [success, setSuccess] = useState(false); // opening snackbar
+
+    // shared toast that displays message for either leaving or adding a course
+    const [toastOpen, setToastOpen] = useState(false);
+    const [toast, setToast] = useState<Toast>({
+        success: true,
+        message: "",
+    });
+
+    const [openLeave, setOpenLeave] = useState(false);
+    const [leaveMembership, setLeaveMembership] = useState<
+        UserMembership | undefined
+    >(undefined);
 
     return (
         <>
             <Grid.Row padded="true">
+                <ModalLeaveStudentCourse
```

Comment:
Would it be better if you wrap this component in a check of whether leaveMembership is undefined? Then you would be able to guarantee that leaveMembership is not undefined in the modal (and you'd change the prop types there to reflect that). The advantage is that the LeaveMembershipModal is easier to read since it can be confusing as to why the membership itself is undefined.

Reply:
Makes sense, then can get rid of all the question marks!